waveshell/pkg/shexec/client.go

```diff
@@ -57,9 +57,26 @@ func (cw CmdWrap) Parser() (*packet.PacketParser, io.ReadCloser, io.ReadCloser,
 }
 
 func (cw CmdWrap) Start() error {
+	defer func() {
+		for _, extraFile := range cw.Cmd.ExtraFiles {
+			extraFile.Close()
```

Comment:
when extraFile is nil, this technical works and just returns an error (https://cs.opensource.google/go/go/+/refs/tags/go1.22.0:src/os/file_posix.go;l=19 -- does a nil check), but feels not so good.  add an `if` here and only call Close if extraFile is not nil.